### PR TITLE
Enhancement/issues#9 add angular setup

### LINKED ISSUE #9: Add Angular2 basic setup to starter template.
Add Angular2 library.
Add basic test
Make sure styles work
Modify webpack configs accordingly.

### PR DESCRIPTION
**Pull request by cxoltero**
resolves #9 

Usually Angular is tested using karma, a few articles show how to have testing done with Jest as well.
[jest & angular2](https://izifortune.com/unit-testing-angular-applications-with-jest/)
this pr uses setup for Jest, open to change it to Karma.

### CODE REVIEW

File: jest.config.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
```
Comment (thescientist13) [inline jest.config.json]: @cxoltero 
This was an exception between the two, but the plan for the Angular seed was to use Karma as I think that's the general use case with Angular, and what their docs generally favor.

cc: @paul-barry-kenzan / @thisgeek

File: webpack.config.common.js [hunk 2]
```diff
@@ -78,6 +76,11 @@ module.exports = {
     new ExtractTextPlugin({
       filename: 'styles.css',
       allChunks: true
-    })
+    }),
+
+    new webpack.ContextReplacementPlugin(
```
Comment (thescientist13) [inline webpack.config.common.js]: what's this for?

Comment (cxoltero) [inline webpack.config.common.js]: when compiling, app loads fine, but it throws a console warning:
```
6857:19-40 Critical dependency: the request of a dependency is an expression
```

this plugin gets rid of it.

File: jest.config.json (hunk 1 above)
Comment (cxoltero) [inline jest.config.json]: :space_invader: :space_invader: cool, will change to karma. thanks

**Pull request closed without merging**